### DISCUSSION TITLE
Unhandled error during request: { message: 'JWT expired' }

### DISCUSSION THREAD

**Discussion by davut** [supabase/supabase#4898] [Questions]
Framework: `Nextjs`,
Hosting: `Vercel`,
Auth: `Supabase`

This is the error message I get
```
JWT expired
2022-01-10T07:24:35.518Z	626879ee-eafb-43cd-930d-591fbc86a04e	ERROR	Unhandled error during request: { message: 'JWT expired' }
2022-01-10T07:24:35.518Z	626879ee-eafb-43cd-930d-591fbc86a04e	ERROR	{ message: 'JWT expired' }
2022-01-10T07:24:35.519Z	626879ee-eafb-43cd-930d-591fbc86a04e	ERROR	Unhandled Promise Rejection 	{"errorType":"Runtime.UnhandledPromiseRejection","errorMessage":"[object Object]","reason":{"message":"JWT expired"},"promise":{},"stack":["Runtime.UnhandledPromiseRejection: [object Object]","    at process.<anonymous> (/var/runtime/index.js:35:15)","    at process.emit (events.js:412:35)","    at processPromiseRejections (internal/process/promises.js:245:33)","    at processTicksAndRejections (internal/process/task_queues.js:96:32)"]}
Unknown application error occurred
```

Hi all,

I could not solve this issue, it might be something else but I started to see it after I added 

```Javascript
supabase.auth.session = () => ({
      access_token: context.req.cookies['sb:token']
});
```
in my getServerSideProps;

I needed it because I wasn't able to pass RLS without it.

After going to the homepage I get automatically signed out and after that I go to the desired page (in my case ```/app/tools/```)  and I don't get that error anymore but when I leave the desired page (in my case ```/app/tools/```) for about an hour and come back and refresh I get that error. 

Here is my getServerSideProps function:

```Javascript
export async function getServerSideProps(context) {


  try {

    supabase.auth.session = () => ({

      access_token: context.req.cookies['sb:token']

    });

  } catch (err) {

    console.log(err);

    throw err;

  }


  let tools = {};

  let categories = {};

  try {

    categories = await getAllCategories();

    tools = await getAllTools();

  } catch (err) {

    console.log(err);

    throw err;

  }


  return {

    props: {

      tools,

      categories

    }

  };

}

```

Here is how I set my cookie in api/auth.js:

```Javascript
export default async (req, res) => {
  supabase.auth.api.setAuthCookie(req, res)
}
```

**Comment by b2m9** [comment]
Supabase doesn't refresh your JWT automatically. Many others have similar issues, see https://github.com/supabase/supabase/discussions/889. You can force a JWT refresh via `supabase.auth.refreshSession`, but it doesn't work all the time for me. There's an open issue to address this problem: https://github.com/supabase/supabase-js/issues/197

**Reply by davut** [reply]
I set `autoRefreshToken: true, persistSession: true` but still getting the same problem were you able to solve this?

I am aware of those issues but the supabase team wanted me to post a discussion about my problem so that's why I am here

**Comment by kangmingtay** [comment]
Hey @davut, just a couple of questions: 1
1. Did you check if the sb:token is being set in the cookie?
2. Are you able to share the RLS policy which is restricting this?

**Reply by davut** [reply]
Hi @kangmingtay;

> 1. Yes, I can see the sb:token when I sign in

> 2. (("right"(email(), 11) = '@example.co'::text) OR (active = true)). policy works fine but when I am requesting in getServerSideProps I can't get the `active = false` when I am authorized as @example.co

**Reply by kangmingtay** [reply]
sry, i might have missed out on something mentioned earlier, but when does `active = false` and what does `active` represent?

**Reply by davut** [reply]
I have `tools` in the database and I want to get all of them if I am authorized as @example.co and if it's the normal user (example -> test@gmail.com) then only get the active users. That's why I use `active`
